### ISSUE TITLE
Key of not selected taxonomy-value appears in catalog-index if it is part of another taxonomy-key.

### ISSUE THREAD

**Issue by adrianschulz** [collective/collective.taxonomy#108]
If a key (e.g. **mas**) is also part of another key (e.g. **mas**terofarts) both values will be indexed if it’s a single-select taxonomy and you select the longer one (e.g. masterofarts).

<h3>Steps to reproduce</h3>

1. Clone from https://github.com/collective/collective.taxonomy.git
2. Pip install & run buildout (I needed to downgrade pycodestyle to 2.5.0)
3. Add the taxonomy (see below) to the default profile of collective.taxonomy.
4. Startup Instance: `bin/instance fg`
5. Install collective.taxonomy 
6. Enable the behavior of the added taxonomy (Datelines taxonomy) for Pages
7. Create a Page and select `Master of Arts`
8. Have a look at how the page is indexed (http://localhost:8080/Plone/portal_catalog/manage_catalogView). Instead of `['masterofarts']` you will find `['mas', 'masterofarts']` in the indexed values.

Tested with commit 14a968e9d85e41cb252b218a4acc898025035d89 

<h3>Taxonomy-Data</h3>

**datelines.cfg**
```
[taxonomy]
name = datelines
title = Datelines Taxonomy
default_language = de
field_title = Datelines Taxonomy
is_single_select = true
```

**datelines.xml**
```
<?xml version='1.0' encoding='utf-8'?>
<vdex xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.imsglobal.org/xsd/imsvdex_v1p0" language="de" orderSignificant="false" profileType="hierarchicalTokenTerms" xsi:schemaLocation="http://www.imsglobal.org/xsd/imsvdex_v1p0 imsvdex_v1p0.xsd http://www.imsglobal.org/xsd/imsmd_rootv1p2p1 imsmd_rootv1p2p1.xsd">
  <vocabName>
    <langstring language="de">Datelines Taxonomy</langstring>
  </vocabName>
  <vocabIdentifier>datelines</vocabIdentifier>
  <term>
    <termIdentifier>masterofarts</termIdentifier>
    <caption>
      <langstring language="de">Master of Arts</langstring>
      <langstring language="en">Master of Arts</langstring>
      <langstring language="fr">Master of Arts</langstring>
    </caption>
  </term>
  <term>
    <termIdentifier>bachelorofarts</termIdentifier>
    <caption>
      <langstring language="de">Bachelor of Science</langstring>
      <langstring language="en">Bachelor of Science</langstring>
      <langstring language="fr">Bachelor of Science</langstring>
    </caption>
  </term>
  <term>
    <termIdentifier>mas</termIdentifier>
    <caption>
      <langstring language="de">MAS</langstring>
      <langstring language="en">MAS</langstring>
      <langstring language="fr">MAS</langstring>
    </caption>
  </term>
</vdex>

```

**Comment by adrianschulz** [comment]
Is anyone having a look at the Pull-Request?

**Comment by adrianschulz** [comment]
Fixing the build: #111 
Fixing the index-bug: #112

**Comment by tomgross** [comment]
Fixed by #112 
I don't have maintainer rights for the package on pypi, so I can't help you with the release unfortunately.
